support/ci/shared.sh

```diff
@@ -1,17 +1,57 @@
 #!/bin/bash
 
-set -eou pipefail
+set -euo pipefail
+
+maybe_install_rustup() {
+  if command -v rustup >/dev/null 2>&1; then
```

Comment:
I kinda like the shorthand version, but it's not a big deal if you prefer the longer form
```suggestion
  if command -v rustup &>/dev/null; then
```

Reply:
I'd add a check for `cargo` here too, since in the event that `rustup` is installed but `"$HOME"/.cargo/env` hasn't been sourced, this function wouldn't handle it correctly.

Reply:
I didn't even know about the shorthand version, thanks!  I like that better too.

I'll add a check for `cargo` as well.